modules/ding_ddbasic/ding_ddbasic.module

```diff
@@ -0,0 +1,373 @@
+    '#default_value' => variable_get('ding_frontpage_events_count', 6),
+  );
+  $form['ding_frontpage']['ding_frontpage_events']['ding_frontpage_events_promoted'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Only promoted events'),
+    '#default_value' => variable_get('ding_frontpage_events_promoted', TRUE),
+  );
+}
+
+/**
+ * Implements hook_pre_build().
+ */
+function ding_ddbasic_views_pre_build(&$view) {
+  switch ($view->name . ' ' . $view->current_display) {
+    case 'ding_event ding_event_list_frontpage':
+
+      if (variable_get('ding_frontpage_events_promoted', TRUE) == FALSE) {
+        unset($view->filter['promote']);
+      }
+
+      $view->display_handler->display->display_options['pager']['options']['items_per_page'] = variable_get('ding_frontpage_events_count', 6);
+      break;
+
+    case 'ding_event ding_event_list':
+      $view->display_handler->options['pager']['options']['items_per_page'] = ddbasic_theme_setting('number_of_events', 10);
+
+      if (ddbasic_theme_setting('group_events_by_date', FALSE)) {
+        $view->field['field_ding_event_date_1']->options['settings']['format_type'] = 'ding_date_only_version2';
+      }
+      break;
+
+    case 'ding_news ding_news_list':
+      $view->display_handler->options['pager']['options']['items_per_page'] = ddbasic_theme_setting('number_of_news', 10);
+
+      break;
+  }
+}
+
+/**
+ * Is the search form extended on the current page.
+ */
+function ding_ddbasic_is_search_form_extended() {
+  return menu_get_item()['path'] !== 'search/ting/%';
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function ding_ddbasic_form_search_block_form_alter(&$form, &$form_state) {
+  if (ding_ddbasic_is_ting_search_extend_form()) {
+    $form['search_field'] = array(
+      '#type' => 'container',
+      '#weight' => 5,
+      '#attributes' => array('class' => array('search-field')),
+      'search_block_form' => $form['search_block_form'],
+      'actions' => $form['actions'],
+    );
+    unset($form['search_block_form'], $form['actions']);
+
+    $form['search_extended_open_button'] = array(
+      '#markup' => '<a href="#" class="search-extended-button">Search extended button</a>',
+    );
+
+    if (ding_ddbasic_is_search_form_extended()) {
+      $form['material_type'] = array(
+        '#type' => 'select',
+        '#options' => array('' => t('Select material')) + drupal_map_assoc(array_keys(_ting_fetch_well_types())),
+        '#weight' => 10,
+        '#default_value' => 'bog',
+      );
+    }
+
+    array_unshift($form['#submit'], 'ding_ddbasic_form_search_block_form_submit');
+  }
+}
+
+/**
+ * Pre submit action for form_search_block_form().
+ */
+function ding_ddbasic_form_search_block_form_submit($form, &$form_state) {
+  if (!empty($form_state['values']['material_type'])) {
+    if (!empty($form_state['values']['search_block_form'])) {
+      $form_state['extra_query_parameters']['facets'][] = 'facet.type:' . $form_state['values']['material_type'];
```

Comment:
We've added some extra code in the ting_search_submit function to handle the "extra_query_parameter" variable. This way we can select the facet instead of writing the filters in the search field.